src/test/java/fr/socrates/domain/meal/MealCateringFeature.java

```diff
@@ -0,0 +1,148 @@
+package fr.socrates.domain.meal;
+
+import fr.socrates.common.Printer;
+import fr.socrates.domain.CandidateId;
+import fr.socrates.domain.attendee.ConfirmationService;
+import fr.socrates.domain.attendee.ConfirmationServiceImpl;
+import fr.socrates.domain.attendee.Payment;
+import fr.socrates.domain.candidate.*;
+import fr.socrates.domain.candidate.Candidate.CandidateBuilder;
+import fr.socrates.domain.checkin.CheckIn;
+import fr.socrates.domain.checkin.CheckInService;
+import fr.socrates.domain.checkin.CheckInServiceImpl;
+import fr.socrates.domain.common.AccommodationChoice;
+import fr.socrates.infra.repositories.InMemoryCandidateRepository;
+import fr.socrates.infra.repositories.InMemoryCheckInRepository;
+import fr.socrates.infra.repositories.InMemoryConfirmationRepository;
+import org.assertj.core.internal.cglib.core.Local;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+import static fr.socrates.domain.meal.Diet.NORMAL;
+import static fr.socrates.domain.meal.Diet.PESCATARIAN;
+import static fr.socrates.domain.meal.Diet.VEGAN;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MealCateringFeature {
+    private static final String JOHN_DOE_FR = "john@doe.fr";
+    private static final String ARTHUR_LEROI_FR = "arthur@leroi.fr";
+    private static final String RAJESH_KOOTRAPOLI_COM = "rajesh@kootrapoli.com";
+    private static final String SHELDON_COOPER_COM = "sheldon@cooper.com";
+    private static final String LEONARD_HOFSTADTER_COM = "leonard@hofstadter.com";
+    private static final String HOWARD_WOLOWITZ_COM = "howard@wolowitz.com";
+    private static final String LEO_DAGAN_FR = "leo@dagan.fr";
+    private static final String KARA_DOC_FR = "kara@doc.fr";
+    private static final String PERSE_VAL_FR = "perse@val.fr";
+    private Candidate johndoe;
+    private Candidate arthurleroi;
+    private Candidate rajeshkootrapoli;
+    private Candidate sheldoncooper;
+    private Candidate leonardhofstadter;
+    private Candidate howardwolowitz;
+    private Candidate leodagan;
+    private Candidate karadoc;
+    private Candidate perceval;
+    private ConfirmationService confirmationService;
+    private MealService mealService;
+
+    @Mock
+    private Printer console;
+
+    @Before
+    public void setUp() throws Exception {
```

Comment:
we could extract methods : we initialize candidates, checkins and we confirm candidates